scripts/kubestellar-snapshot.sh

```diff
@@ -0,0 +1,580 @@
+# Define the echov function based on verbosity
+if [ "$arg_verbose" == "true" ]; then
+    echov() { echo "$@" ; }
+else
+    echov() { :; }
+fi
+
+
+###############################################################################
+# Script info
+###############################################################################
+echov -e "${COLOR_INFO}${SCRIPT_NAME} v${SCRIPT_VERSION}{COLOR_NONE}\n"
+echov -e "Script run on ${COLOR_INFO}$TIMESTAMP${COLOR_NONE}"
+
+
+###############################################################################
+# Check dependencies
+###############################################################################
+echov "Checking script dependencies:"
+
+is_installed 'kubectl' \
+    'kubectl' \
+    'kubectl version --client | head -1 | cut -d" " -f3'
+
+is_installed 'helm' \
+    'helm' \
+    'helm version | cut -d"\"" -f2'
+
+is_installed 'jq' \
+    'jq' \
+    'jq --version'
+
+
+###############################################################################
+# Ensure output folder
+###############################################################################
+if [[ "$arg_logs" == "true" || "$arg_yaml" == "true" ]] ; then
+    mkdir -p "$OUTPUT_FOLDER"
+fi
+
+
+###############################################################################
+# Determine the list of kubeconfigs to search
+# based on https://kubernetes.io/docs/reference/k/generated/k_config/
+###############################################################################
+[[ -n "$arg_kubeconfig" ]] && export KUBECONFIG="$arg_kubeconfig"
+[[ -z "$KUBECONFIG" ]] && export KUBECONFIG="$HOME/.kube/config"
+echov -e "Using kubeconfig(s): ${COLOR_INFO}$KUBECONFIG${COLOR_NONE}"
+
+
+###############################################################################
+# Determine the list of contexts to search
+###############################################################################
+if [[ "$arg_context" == "" ]] ; then
+    contexts=($(kubectl config get-contexts --no-headers -o name))
+else
+    contexts=("$arg_context")
+fi
+
+if [[ -z "${contexts[@]}" ]] ; then
+    echoerr "No context(s) found in the kubeconfig file $KUBECONFIG!"
+    exit 1
+fi
+
+echov "Validating contexts(s): "
+valid_contexts=()
+for context in "${contexts[@]}" ; do # for all contexts
+    [[ -z "$context" ]] && continue
+    [[ "$context" == "$(kubectl config current-context || true)" ]] && cc="*" || cc=""
+    if kubectl --context $context get secrets -A > /dev/null 2>&1 ; then
+        echov -e "${COLOR_GREEN}\xE2\x9C\x94${COLOR_NONE} ${COLOR_INFO}${context}${COLOR_NONE} ${COLOR_GREEN}$cc${COLOR_NONE}"
+        [[ -z "$cc" ]] && valid_contexts+=("$context") || valid_contexts=("$context" "${valid_contexts[@]}")
+    else
+        echov -e "${COLOR_RED}X${COLOR_NONE} ${COLOR_INFO}${context}${COLOR_NONE} ${COLOR_GREEN}$cc${COLOR_NONE}"
+    fi
+done
+
+if [[ -z "${valid_contexts[@]}" ]] ; then
+    echoerr "No valid context(s) found in the kubeconfig file $KUBECONFIG!"
```

Comment:
FYI, I find the wording "valid context(s)" to be a bit misleading. "Validity" of contexts is something that I expect the unwashed user to suppose is defined by Kubernetes. I think it would be better to say something like "No KubeStellar core installation found in any context in the kubeconfig file $KUBECONFIG".

Reply:
This is not related to KubeStellar, it's trying to remove configs from the list that are not working.
If you hit this erro, there is nowhere to search

Reply:
So something like "broken" or "malfunctioning" would be more on-point than "[not] valid". "valid" often means something more shallow about syntax and self-consistency. Not a big deal.